content/exchange/artifacts/IRIS.Timeline.Add.yaml

```diff
@@ -0,0 +1,219 @@
+  - name: AddToSummary
+    description: Add it to timeline summary?
+    type: bool
+  - name: AddToGraph
+    description: Add it to attack graph?
+    type: bool
+  - name: Category
+    description: "Category of the action, mostly MITRE Enterprise Tactics. Allowed options are abbreviations and their MITRE ID: tbd,legit,rem,ini,exec,pers,priv,def,creds,disc,lat,coll,c2,exfil,imp"
+    type: choices
+    choices:
+      - tbd
+      - legit
+      - rem
+      - ini
+      - exec
+      - pers
+      - priv
+      - def
+      - creds
+      - disc
+      - lat
+      - coll
+      - c2
+      - exfil
+      - imp
+  - name: Color
+    description: Specify the color for this event in Iris. Green by default for obvious reasons.
+    type: choices
+    choices:
+      - green
+      - white
+      - blue
+      - lightblue
+      - purple
+      - red
+      - orange
+  - name: RawEvent
+    description: Add the raw event, message or the entire row as additional information. 
+  - name: IrisURL
+    description: URL of DFIR-IRIS. Preferred method is to use the server metadata
+  - name: IrisKey
+    description: API Key of DFIR-IRIS. Preferred method is to use the server metadata
+  - name: IrisCaseId
+    description: Case ID of the current case. Preferred method is to use the server metadata
+  - name: IrisRootCA
+    description: RootCA of DFIR-IRIS for self-signed or internal certificates of DFIR-IRIS. Preferred over completely skipping SSL verification.
+  - name: DisableSSLVerify
+    default: false
+    description: Disable TLS verification for HTTPS request to DFIR-IRIS.
+
+sources:
+
+  - query: |
+
+      LET URL <= if(
+            condition=IrisURL,
+            then=IrisURL,
+            else=server_metadata().IrisURL)
+      LET Creds <= if(
+           condition=IrisKey,
+           then=IrisKey,
+           else=server_metadata().IrisKey)
+      LET CaseID <= if(
+           condition=IrisCaseId,
+           then=IrisCaseId,
+           else=server_metadata().IrisCaseId)
+      LET RootCA <= if(
+            condition=IrisRootCA,
+            then=IrisRootCA,
+            else=server_metadata().IrisRootCA)
+            
+      LET metadata_preparation = SELECT client_metadata(client_id=clientId) as metadata FROM scope() WHERE metadata.IRIS_AssetId
+      
+      LET syncAsset = SELECT * FROM Artifact.Custom.Exchange.IRIS.Sync.Asset(clientId=clientId,IrisURL=URL,IrisCaseId=CaseID,IrisKey=Creds,IrisRootCA=RootCA,DisableSSLVerify=DisableSSLVerify)
+        
+      LET eventAsset1 = if(condition=metadata_preparation,then=array(a=metadata_preparation.metadata[0].IRIS_AssetId),else=if(condition=syncAsset.Result[0]="SUCCESS",then=array(a=metadata_preparation.metadata[0].IRIS_AssetId),else=[]))
+        
+      LET eventAsset2 = if(condition=AdditionalAssetId,then=split(string=AdditionalAssetId,sep=",|;"),else=[])
+        
+      LET eventCategory = if(condition=Category=~"^legit",then=2,
```

Comment:
Just fyi this type of thing can now be done easier with regex_transform()

Reply:
Hmm, yep, most of the cascaded if statements can be easier implemented using the regex_transform function. However, there's no possibility for some sort of a final 'else' statement or a default/catch all.

Reply:
If the Category is a fixed string you can just use a lookup to sub it like this for example https://github.com/Velocidex/SQLiteHunter/blob/55607603acc357becb592d522a2430e6ed86901e/definitions/ChromiumBrowser_HistoryVisits.yaml#L55